### PR TITLE
Return most recent results first from findDatesUTC

### PR DESCRIPTION
**Pull request by sinergise-anze**
Service `findAvailableDates` is not consistent in the order in which the results are returned, so we check and reverse them if needed (we assume that the order is either `leastRecent` or `mostRecent`, not random).

### CODE REVIEW

**Comment by guss84** [comment]
I am pretty sure the order wont be random, but wouldnt it be safer to just sort the list and not make the assumption that the order is not random?

> Service `findAvailableDates` is not consistent in the order in which the results are returned, so we check and reverse them if needed (we assume that the order is either `leastRecent` or `mostRecent`, not random).

I am pretty sure the order wont be random, but wouldnt it be safer to just sort the list and not make the assumption that the order is not random?

File: src/layer/AbstractSentinelHubV3Layer.ts [hunk 1]
```diff
@@ -233,6 +233,16 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
       ...this.getFindDatesUTCAdditionalParameters(),
     };
     const response = await axios.post(this.dataset.findDatesUTCUrl, payload);
-    return response.data.map((date: string) => moment.utc(date).toDate());
+    const result = response.data.map((date: string) => moment.utc(date).toDate());
+
+    // S-5P, S-3 and possibly other datasets return the results in reverse order (leastRecent).
```
Comment (dgostencnik) [inline src/layer/AbstractSentinelHubV3Layer.ts:238]: This is also the case with byoc datasets. I'm not sure about sorting. It would probably be more clear and readable, but on the other hand in most of the cases it'll be just sorting already sorted array.

Comment (sinergise-anze) [reply src/layer/AbstractSentinelHubV3Layer.ts:238]: This was my assumption too, but @guss84 is right, I should check that this is true... I'll dig up this information and post it here.

Comment (sinergise-anze) [reply src/layer/AbstractSentinelHubV3Layer.ts:238]: It seems that this is a bug on the service side (I filed an issue), however it might take some time for this to be fixed. 

I have also noticed that `BYOCLayer` was re-implementing the same logic (but used `await` for `getFindDatesUTCAdditionalParameters`), so I fixed `getFindDatesUTCAdditionalParameters` to be `async` (no harm if function returns an object instead of Promise). 

I have also checked all other layers and added sorting to `WmsLayer`, so now `findDatesUTC` should always return sorted data, no matter which layer it is called on.

File: src/layer/AbstractSentinelHubV1OrV2Layer.ts [hunk 2]
```diff
@@ -115,7 +115,7 @@ export class AbstractSentinelHubV1OrV2Layer extends AbstractLayer {
     };
   }
 
-  protected getFindDatesUTCAdditionalParameters(): Record<string, any> {
+  protected async getFindDatesUTCAdditionalParameters(): Promise<Record<string, any>> {
```
Comment (dgostencnik) [inline src/layer/AbstractSentinelHubV1OrV2Layer.ts:118]: Just curious why was this changed to async? fftr

Comment (sinergise-anze) [reply src/layer/AbstractSentinelHubV1OrV2Layer.ts:118]: Cleanup - `BYOCLayer` already defined it as `async` (because it needs to call `await this.fetchLayerParamsFromSHServiceV3`) and it's better to be consistent. 

This meant that `findDatesUTC` code was copy-pasted to `BYOCLayer` just to add an `await`... Since there is no harm in making this function `async`, I have simply changed the default signature and cleaned up duplicated code from `BYOCLayer`.
